dokka-runners/dokka-gradle-plugin/src/testFunctional/kotlin/KmpCommonSourceSharedWithDependentsTest.kt

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2014-2024 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+package org.jetbrains.dokka.gradle
+
+import io.kotest.assertions.withClue
+import io.kotest.core.annotation.Ignored
+import io.kotest.core.spec.style.FunSpec
+import io.kotest.matchers.sequences.shouldBeEmpty
+import org.gradle.testkit.runner.TaskOutcome.*
+import org.jetbrains.dokka.gradle.internal.DokkaConstants.DOKKA_VERSION
+import org.jetbrains.dokka.gradle.utils.*
+import kotlin.io.path.isRegularFile
+import kotlin.io.path.useLines
+import kotlin.io.path.walk
+
+@Ignored("KMP: References is not linked if they are in shared code and there is an intermediate level between them https://github.com/Kotlin/dokka/issues/3382")
+class KmpCommonSourceSharedWithDependentsTest : FunSpec({
```

Comment:
I believe this test should be placed into `dokka-integration-tests` so that we will be able to check it with different Kotlin versions.
In my understanding, `functional tests` in DGPv2 are more about plugin behavior while this test is more about overall Dokka behavior and so should be in IT

Reply:
Good idea, done.

Could you check the expected data in `dokka-integration-tests/gradle/projects/it-kotlin-multiplatform/expectedData/html`? There are errors, but I'm not sure if any are DGP's responsibility, or are caused by https://github.com/Kotlin/dokka/issues/3382

Reply:
I've checked `expectedData/html` and it looks reasonable and the error is most likely caused by #3382.
let's ignore test and re-check it in scope of https://github.com/Kotlin/dokka/issues/3930 (DGPv2 + K2)

Reply:
Hmmmmm it looks like locally I get `Error type: Unresolved type for NativeMainCls`, but not on TeamCity?? Why??

https://ge.jetbrains.com/s/2cdhjngaftnhk/tests/task/:dokka-integration-tests:gradle:test/details/org.jetbrains.dokka.it.gradle.KotlinMultiplatformIT/generate%20dokka%20HTML(DokkaGradleProjectRunner)%5B1%5D?focused-exception-line=2-100&top-execution=4

Reply:
The difference in OS (linux vs macOS) and so probably the difference is in dependencies passed inside KGP. E.g `iOS` targets are not compilable on linux and so probably the classpath (or some other configuration) is somehow different, which causes this behavior.

In this case, I think that for this test probably having just `linux` targets should be enough to reproduce the issue, correct?
We could also try to add `mingw` target (or `androidNative` targets), or add additional intermediate sourceSets somewhere, so that all targets will be cross-compiled on all OSs.

Reply:
I could understand it if locally, on my Mac, I would get no errors and on Linux the macOS targets would cause `Unresolved type` errors, but I don't see why it's the other way around. I'd like to understand the problem more before trying to work around it, but I'm not sure how.

I'll check with my team.